app/models/gift.rb

```diff
@@ -0,0 +1,18 @@
+class Gift < ApplicationRecord
+  has_many :gift_categorizations
+  has_many :categories, through: :gift_categorizations
+  belongs_to :supplier
+  delegate :name, to: :supplier, prefix: true
+  has_one_attached :image
+
+  scope :get_gifts_from_categories, lambda { |categories, order|
+    with_attached_image.includes(:supplier)
+                       .joins(:gift_categorizations)
+                       .where(gift_categorizations: { category_id: categories })
+                       .order(order).distinct
+  }
```

Comment:
This is a weird scope for the following reasons:
- The name should be something like `with_categories`, so that we can read it like `Gift.with_categories(ids)`
- It does too many things: it should only do what the name says, join it with categories and filter by category id. Let's remove the includes and the `with_attached_image` scope. Remember that you can chain scopes, so we can do
```rb
Gift.with_categories(ids).with_attached_image.includes(:supplier).order(order).distinct
```